spec.md

```diff
@@ -458,7 +457,32 @@ $ curl http://username:password@broker-url/v2/service_instances/:instance_id?acc
 | 409 Conflict | MUST be returned if a service instance with the same id already exists but with different attributes. The expected response body is `{}`, though the description field MAY be used to return a user-facing error message, as described in [Broker Errors](#broker-errors). |
 | 422 Unprocessable Entity | MUST be returned if the broker only supports asynchronous provisioning for the requested plan and the request did not include `?accepts_incomplete=true`. The expected response body is: `{ "error": "AsyncRequired", "description": "This service plan requires client support for asynchronous service operations." }`, as described below. |
 
-Responses with any other status code will be interpreted as a failure. Brokers can include a user-facing message in the `description` field; for details see [Broker Errors](#broker-errors).
+Responses with any other status code will be interpreted as a failure, and in some cases, the broker client SHOULD initiate orphan mitigation. See the next section for details on orphan mitigation. Also, brokers can include a user-facing message in the `description` field; for details see [Broker Errors](#broker-errors).
+
+##### Orphans
+
+The platform marketplace is the source of truth for service instances. Service brokers are expected to have successfully provisioned all the service instances and bindings that the marketplace knows about, and none that it doesn't.
+
+Orphans can result if the broker does not return a response before a provision request from the marketplace times out (typically 60 seconds). For example, if a broker does not return a response to a provision request before the request times out, the broker might eventually succeed in provisioning a service instance after the marketplace considers the request a failure. This results in an orphan service instance on the broker's side.
+
+To mitigate orphan service instances and bindings, the marketplace SHOULD attempt to deprovision resources it cannot be sure were successfully created, and SHOULD keep trying to delete them until the broker responds with a success.
+
+Platforms SHOULD initiate orphan mitigation in the following scenarios:
+
+| Status code of broker response | Platform interpretation of response | Platform initiates orphan mitigation? |
+| --- | --- | --- |
+| 200 | Success | No |
+| 200 with malformed response | Failure | No |
+| 201 | Success | No |
+| 201 with malformed response | Failure | Yes |
+| All other 2xx | Failure | Yes |
+| 408 | Failure due to timeout | Yes |
+| All other 4xx | Broker rejected request | No |
+| 5xx | Broker error | Yes |
+| Timeout | Failure | Yes |
+
+If the platform marketplace encounters an internal error provisioning a service instance or binding (for example, saving to the database fails), then it MUST at least send a single delete or unbind request to the service broker to prevent creation of an orphan.
```

Comment:
This sentence is ambiguous and confusing.
"it MUST at least send a **single** delete or unbind request" - what does it mean? Sounds like "at least I tried" excuse to me. What if this request doesn't reach service broker due to network failure, for example? What if marketplace crashed right after initiating provisioning? If there is no guaranteed persistence on marketplace side, it simply cannot guarantee to fulfill this **MUST** requirement (or, in other words, "at-least once" semantics).

I'd rather require marketplace to save the binding to the database **before** it makes the first call to service broker for provisioning. Given that marketplace is considered the *source of truth*, infrastructure failure should not be an excuse for running into out-of-sync state.